### ISSUE TITLE
Can't reconstruct sparse/dense model from known camera poses

### ISSUE THREAD

**Issue by Lee-JaeWon** [colmap/colmap#2309]
When i know the pose for each image, how do i configure the sparse model?
I referred to ['Reconstruct sparse/dense model from known camera photos'](https://colmap.github.io/faq.html#reconstruct-sparse-dense-model-from-known-camera-poses) in the document, but I don't understand the content.

`images.txt` and `camera.txt` were constructed, and `points3D.txt` were left blank.
I don't know what to do here.

It would be nice if you could tell me in detail.

Thank you for your wonderful work.

#497 #1647

**Comment by Lee-JaeWon** [comment]
@tsattler Should I create `database.db` file from `images.txt` and `camera.txt`?

In colmap gui, i performed feature extraction, feature matching, and start reconstruction,
How should this be done if you know the pose?

**Comment by tsattler** [comment]
Yes, you will need to create a database and the images.txt, cameras.txt, and points3d.txt files. The database needs to contain features and feature matches.
If you already have a database, e.g., created using the gui, you can use it.

Once you have these files, you call `colmap point_triangulator` from the command line.

**Comment by longyangqi** [comment]
> Yes, you will need to create a database and the images.txt, cameras.txt, and points3d.txt files. The database needs to contain features and feature matches. If you already have a database, e.g., created using the gui, you can use it.
> 
> Once you have these files, you call `colmap point_triangulator` from the command line.

Hello，when I do the 'colmap point_triangulator' with command:
>colmap point_triangulator --database_path $DATASET_PATH/database.db \
    --image_path $DATASET_PATH/images \
    --input_path $DATASET_PATH/sparse_created \
    --output_path $DATASET_PATH/sparse

i get errors, the log is:

Feature matching
I0911 22:54:59.458237 114219 sift.cc:1423] Creating SIFT GPU feature matcher
I0911 22:54:59.863863 114214 pairing.cc:168] Generating exhaustive image pairs...
I0911 22:54:59.863891 114214 pairing.cc:201] Matching block [1/2, 1/2]
I0911 22:55:01.089156 114214 feature_matching.cc:46] in 1.225s
I0911 22:55:01.095078 114214 pairing.cc:201] Matching block [1/2, 2/2]
I0911 22:55:01.192881 114214 feature_matching.cc:46] in 0.098s
I0911 22:55:01.198748 114214 pairing.cc:201] Matching block [2/2, 1/2]
I0911 22:55:01.611954 114214 feature_matching.cc:46] in 0.413s
I0911 22:55:01.617528 114214 pairing.cc:201] Matching block [2/2, 2/2]
I0911 22:55:01.694736 114214 feature_matching.cc:46] in 0.077s
I0911 22:55:01.699519 114214 timer.cc:91] Elapsed time: 0.039 [minutes]
I0911 22:55:01.830173 114270 misc.cc:198] 
Loading model
I0911 22:55:02.385186 114270 incremental_mapper.cc:232] Loading database
I0911 22:55:02.482928 114270 database_cache.cc:65] Loading cameras...
I0911 22:55:02.494025 114270 database_cache.cc:75]  65 in 0.011s
I0911 22:55:02.494076 114270 database_cache.cc:83] Loading matches...
I0911 22:55:02.520272 114270 database_cache.cc:89]  326 in 0.026s
I0911 22:55:02.520288 114270 database_cache.cc:105] Loading images...
I0911 22:55:02.921326 114270 database_cache.cc:155]  65 in 0.401s (connected 65)
I0911 22:55:02.921373 114270 database_cache.cc:166] Building correspondence graph...
I0911 22:55:02.943890 114270 database_cache.cc:195]  in 0.023s (ignored 0)
I0911 22:55:02.943972 114270 timer.cc:91] Elapsed time: 0.008 [minutes]
I0911 22:55:02.961793 114270 incremental_mapper.cc:550] Iterative triangulation
I0911 22:55:02.961809 114270 incremental_mapper.cc:556] Triangulating image #59 (0)
I0911 22:55:02.961820 114270 incremental_mapper.cc:558] => Image sees 0 / 1416 points
E0911 22:55:02.961838 114270 image.h:174] 'camera_ptr_' Must be non NULL
terminate called after throwing an instance of 'std::invalid_argument'
  what():  [image.h:174] 'camera_ptr_' Must be non NULL
*** Aborted at 1726066502 (unix time) try "date -d @1726066502" if you are using GNU date ***
PC: @     0x7ff7cf37d00b gsignal
*** SIGABRT (@0x1be5e) received by PID 114270 (TID 0x7ff7cade5000) from PID 114270; stack trace: ***
    @     0x7ff7d0b84631 (unknown)
    @     0x7ff7d0b5b420 (unknown)
    @     0x7ff7cf37d00b gsignal
    @     0x7ff7cf35c859 abort
    @     0x7ff7cf7348d1 (unknown)
    @     0x7ff7cf74037c (unknown)
    @     0x7ff7cf7403e7 std::terminate()
    @     0x7ff7cf740699 __cxa_throw
    @     0x559ef2b45bb9 _ZN6colmap23IncrementalTriangulator16TriangulateImageERKNS0_7OptionsEj.cold
    @     0x559ef2cdd17a colmap::IncrementalMapper::TriangulateImage()
    @     0x559ef2c68136 colmap::IncrementalPipeline::TriangulateReconstruction()
    @     0x559ef2b97624 colmap::RunPointTriangulatorImpl()
    @     0x559ef2b97c06 colmap::RunPointTriangulator()
    @     0x559ef2b788f4 main
    @     0x7ff7cf35e083 __libc_start_main
    @     0x559ef2b7fa4e _start
Aborted (core dumped)

I want to know how to fix it. 
Many thanks!

**Comment by ahojnnes** [comment]
Closing: point_triangulator with known poses explained by maintainer.
